test/replication-luatest/gh_6033_box_promote_demote_test.lua

```
@@ -0,0 +1,505 @@
+    }
+
+    g.server_1 = g.cluster:build_and_add_server(
+        {alias = 'server_1', engine = engine, box_cfg = g.box_cfg})
+    g.server_2 = g.cluster:build_and_add_server(
+        {alias = 'server_2', engine = engine, box_cfg = g.box_cfg})
+    g.cluster:start()
+end
+
+local cluster_reinit = function(g)
+    g.server_1:wait_lsn(g.server_2)
+    g.server_2:wait_lsn(g.server_1)
+
+    g.server_1:box_config(g.box_cfg)
+    g.server_2:box_config(g.box_cfg)
+end
+
+g_nspaces.before_all(cluster_init)
+g_spaces.before_all(cluster_init)
+
+g_nspaces.after_all(function(g) g.cluster:drop() end)
+g_spaces.after_all(function(g) g.cluster:drop() end)
+
+g_nspaces.after_each(cluster_reinit)
+g_spaces.after_each(cluster_reinit)
+
+-- Promoting/demoting should succeed if server is not configured.
+g_unconfigured.test_unconfigured = function()
+    local ok, err = pcall(box.ctl.promote)
+    luatest.assert(ok, string.format(
+        'error while promoting unconfigured server: %s', err))
+
+    local ok, err = pcall(box.ctl.demote)
+    luatest.assert(ok, string.format(
+        'error while demoting unconfigured server: %s', err))
+end
+
+-- Promoting current raft leader and synchro queue owner should succeed
+-- with elections enabled.
+g_nspaces.test_leader_promote = function(g)
+    g.server_1:box_config({election_mode = 'manual'})
+    g.cluster:promote(g.server_1)
+
+    local ok, err = g.server_1:exec(function()
+        return pcall(box.ctl.promote)
+    end)
+    luatest.assert(ok, string.format(
+        'error while promoting leader with elections on: %s', err))
+
+    g.server_1:box_config({election_mode = 'off'})
+    local ok, err = g.server_1:exec(function()
+        return pcall(box.ctl.promote)
+    end)
+    luatest.assert(ok, string.format(
+        'error while promoting leader with elections off: %s', err))
+end
+
+g_nspaces.after_test('test_leader_promote', function(g)
+    g.cluster:demote(g.server_1)
+    g.server_2:wait_synchro_queue_owner_id(0)
```

Why can't these `after_test` just be moved into the test's body? Same in all the other places. IMO these are not kind of tests which can restore their state after any moment of a test case. It is not just a space removal or something. I would just let it fail with all the colors in case of a bug instead of complicating the code.
Up to you.